### DISCUSSION TITLE
Horribly slow, cripplingly slow!

### DISCUSSION THREAD

**Discussion by Denebet** [quodlibet/quodlibet#4535] [General]
Steps to reproduce
------------------

- Launch QL
- switch from the "Paned browser" view to the "Playlists" view
- etc.

Expected Output
---------------

React almost instantly like Tauon Music Box (also developed in python)

Actual Output
-------------

Just to launch QL: around 30 seconds, compared to 3 for Tauon Music Box, or around 10 times more.

Test System
-----------

### Which version of Quod Libet?

4.7 (development)
### Which operating system 

Ubuntu 24.04

### If it's audio-related, what back-end?

Additional Information
----------------------

**Comment by FerdiScholten** [comment]
I cannot reproduce this on my Ubuntu 24.04
starting QL takes at most 2.5 seconds (when starting with the cover grid which is by far the slowest to load) Other browsers 1 second or less.
switching between browsers is almost instantaneous.
This is with 18240 songs in the library on a 6 year old Kabylake based laptop with Intel core I7-8650U 4core processor.

**Comment by declension** [comment]
I also can't reproduce this at all (although on a fast machine with ~9000 tracks)

**Comment by Denebet** [comment]
My context:
- Ubuntu on Xorg;
- AMD Ryzen™ 5 5600G with Radeon™ Graphics × 12 on ASUSTeK COMPUTER INC. PRIME B450-PLUS with 16.0 GiB RAM assembled in 2023;
- collection of 50,000 tracks, most in FLAC format, on Crucial 2TB external SSD connected directly to the motherboard via USB-C.

Opening QL is sometimes very quick (a few seconds at most), but most often takes more than 20 seconds, while launching Tauon Music Box is almost instantaneous. "Scan Library" takes many minutes, switching from one view to another many seconds, software hangs, and quite often crashes and, on restart, message to automatically send a bug report.
As I am not a developer or IT professional, can you tell me some tests and reports to submit to help diagnose the problem?

**Comment by FerdiScholten** [comment]
Forgot to mention that my library has two locations, roughly 4000 songs are on my laptop itself (SSD), the others are accessed over the network on my NAS. QL is fast with both.
Music is in various formats a lot of it in FLAC, but also mp3, wav and even some WMA. 

Makes me think that there is an IO bottleneck on your system somewhere. Or you might be affected by the AMD fTPM bug which also causes lag and stutter.

**Comment by Denebet** [comment]
According to https://www.asus.com/fr/motherboards-components/motherboards/prime/prime-b450-plus/helpdesk_bios?model2Name=PRIME-B450-PLUS the fTPM problem had been resolved long before purchasing my motherboard. And Tauon Music Box, also written in python, does not have this problem.
I just updated the BIOS anyway, but I don't see any difference.
Launching from the terminal returns absolutely nothing.

**Reply by FerdiScholten** [reply]
Start QL with the --debug parameter to get a ton of output in your terminal.

**Comment by Denebet** [comment]
THANKS !
I just observed 8 seconds delay for launching QL which is almost fast compared to other times.
With
$./quodlibet/quodlibet.py --debug
I don't notice any error messages or warnings, but, surprisingly variable, some very long delays when adding tracks to a playlist, example:
D: [0000.000] XSPFBackedPlaylist._emit_changed: Changed playlist <quodlibet.util.collection.XSPFBackedPlaylist object at 0x7b484004bd10> (remove_songs)
on 8,412 tracks took 1'30'' (almost instantaneous with Tauon Music Box).
Just removing a track from a playlist takes about 6''.
I just noticed a 16 second start while Tauon was closed (I specify this because I suspect Tauon of making a mess).
